apis/nucleus/src/components/listbox/ListBoxPortal.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import ListBoxInline from './ListBoxInline';
+
+export default function ListBoxPortal({ app, fieldIdentifier, stateName, element, options }) {
+  let returnComponent = null;
+  const isFrequencyMaxNeeded = options.histogram || options.frequencyMode !== 'N';
+  if (fieldIdentifier.qLibraryId && isFrequencyMaxNeeded) {
+    returnComponent = (
+      <ListBoxFetchMasterItem
+        app={app}
+        fieldIdentifier={fieldIdentifier}
+        stateName={stateName}
+        element={element}
+        options={options}
+      />
+    );
+  } else {
+    returnComponent = (
+      <ListBoxInline
+        app={app}
+        fieldIdentifier={fieldIdentifier}
+        stateName={stateName}
+        element={element}
+        options={options}
+      />
+    );
+  }
+
+  return ReactDOM.createPortal(returnComponent, element);
+}
+
+export function ListBoxFetchMasterItem({ app, fieldIdentifier, stateName = '$', options = {} }) {
+  const [fieldDef, setFieldDef] = useState('');
+
+  useEffect(() => {
+    async function fetchData() {
+      const dim = await app.getDimension(fieldIdentifier.qLibraryId);
+      const dimLayout = await dim.getLayout();
+      setFieldDef(dimLayout.qDim.qFieldDefs ? dimLayout.qDim.qFieldDefs[0] : '');
+    }
```

Comment:
Do we need to handle (or act) on errors? Showing error message may be handled automatically but the question is whether the Listbox will refresh/fetch again and if it matters for this specific feature.

Reply:
Good point, it doesn't seem to fetch again but I will display the error message to the user.

Reply:
The error message should be handled by Engine (if it's not, it should be added there I think).
Maybe add a variable that keeps track of whether data has been fetched or not 🤔 And as long as that variable is `false` (i.e. data not fetched successfully), then disable the feature that depends on the data.